public/app/plugins/datasource/elasticsearch/query_builder.ts

```diff
@@ -94,6 +95,26 @@ export class ElasticQueryBuilder {
     return queryNode;
   }
 
+  getNestedAgg(aggDef: Nested, queryNode: any, target: ElasticsearchQuery) {
+    if (!aggDef.field || !target || !target.metrics) {
```

Comment:
@gabor can we use a better type of `queryNode` where? I see another definition using `{ terms?: any; aggs?: any }` which is also not great.

Reply:
> @gabor can we use a better type of `queryNode` here? I see another definition using `{ terms?: any; aggs?: any }` which is also not great.

i agree. in general, the type-coverage in `query_builder.ts` is not great, so i'm worried that nicely-typing this thing could cause a lot of extra work all around the file, so i'd focus on how much it can be improved in this function locally.. something like  `{nested: {path:string}, aggs: {....}`?